data/maps/SlateportCity_BattleTentBattleRoom/scripts.inc

```diff
@@ -58,41 +61,40 @@ SlateportCity_BattleTentBattleRoom_EventScript_2099BE:: @ 82099BE
 	special DoSpecialTrainerBattle
 	waitstate
 	switch VAR_RESULT
-	case 1, SlateportCity_BattleTentBattleRoom_EventScript_209A39
-
-SlateportCity_BattleTentCorridor_EventScript_209A1B:: @ 8209A1B
+	case 1, SlateportCity_BattleTentBattleRoom_EventScript_DefeatedOpponent
+SlateportCity_BattleTent_EventScript_WarpToLobbyLost:: @ 8209A1B
 	frontier_set FRONTIER_DATA_CHALLENGE_STATUS, CHALLENGE_STATUS_LOST
 	special LoadPlayerParty
 	warp MAP_SLATEPORT_CITY_BATTLE_TENT_LOBBY, 255, 6, 6
 	waitstate
+@ forced stop
```

Comment:
Are you sure these `warp` + `waitstate` combos stop the script?  From a quick look at the source code, I couldn't see why that would happen.

Reply:
Not sure how, but I've seen enough of them that wouldn't make sense if execution continued. Take for example `VerdanturfTown_BattleTentBattleRoom_EventScript_WarpToLobbyLost`, where if it doesn't stop it continues right into `VerdanturfTown_BattleTentBattleRoom_EventScript_DefeatedOpponent`, which is entirely the opposite of what the previous script was doing.